Cargo.toml

```diff
@@ -78,18 +78,19 @@ runtime-tokio = ["_rt-tokio", "sqlx-core/_rt-tokio", "sqlx-macros?/_rt-tokio"]
 
 # TLS features
 tls-native-tls = ["sqlx-core/_tls-native-tls", "sqlx-macros?/_tls-native-tls"]
-tls-rustls = ["sqlx-core/_tls-rustls", "sqlx-macros?/_tls-rustls"]
+tls-rustls-aws-lc-rs = ["sqlx-core/_tls-rustls-aws-lc-rs", "sqlx-macros?/_tls-rustls-aws-lc-rs"]
+tls-rustls-ring = ["sqlx-core/_tls-rustls-ring", "sqlx-macros?/_tls-rustls-ring"]
```

Comment:
Comment from the sidelines:
I know that currently no such checks exist, but would it make sense to have these features be mutually exclusive?
I don't think having two enabled be intentional, or am I missing something?

```rust
#[cfg(all(feature = "tls-rustls-aws-lc-rs", feature = "tls-rustls-ring"))]
compile_error!("feature \"tls-rustls-aws-lc-rs\" and feature \"tls-rustls-ring\" specify crypto providers. Please make a concious choice between either provider");
```

Also, documenting the flags [here](https://github.com/launchbadge/sqlx?tab=readme-ov-file#install) is likely something that would be benefitial

Reply:
Having mutually exclusive features causes all sorts of problems due to [feature unification](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification).

A transitive dependency enabling the wrong feature can break the build, and it can be incredibly frustrating for the end-user, often requiring `[patch]` overrides to remove the offending feature.

We used to have this exact problem with the `runtime-*` features and that was fixed by creating our own facade that automatically chooses a runtime internally. We still sort of have this problem with the `time` and `chrono` features with the macros that I'm trying to fix in #3383.

Friends don't let friends create mutually exclusive features.

Reply:
@djc however, I would retain `tls-rustls` as an alias for `tls-rustls-ring` to not be a breaking change. The user can always change it or pick the more explicit feature instead.